src/ui/public/vislib/visualizations/point_series.js

```diff
@@ -0,0 +1,230 @@
+      // are still a part of the dataset. We want to start the endzones just
+      // outside of them so we will use these values rather than ordered.min/max
+      const oneUnit = (ordered.units || _.identity)(1);
+
+      // points on this axis represent the amount of time they cover,
+      // so draw the endzones at the actual time bounds
+      const leftEndzone = {
+        x: isHorizontal ? 0 : Math.max(xScale(ordered.min), 0),
+        w: isHorizontal ? Math.max(xScale(ordered.min), 0) : height - Math.max(xScale(ordered.min), 0)
+      };
+
+      const rightLastVal = xAxis.expandLastBucket ? ordered.max : Math.min(ordered.max, _.last(xAxis.values));
+      const rightStart = rightLastVal + oneUnit;
+      const rightEndzone = {
+        x: isHorizontal ? xScale(rightStart) : 0,
+        w: isHorizontal ? Math.max(width - xScale(rightStart), 0) : xScale(rightStart)
+      };
+
+      this.endzones = svg.selectAll('.layer')
+        .data([leftEndzone, rightEndzone])
+        .enter()
+        .insert('g', '.brush')
+        .attr('class', 'endzone')
+        .append('rect')
+        .attr('class', 'zone')
+        .attr('x', function (d) {
+          return isHorizontal ? d.x : 0;
+        })
+        .attr('y', function (d) {
+          return isHorizontal ? 0 : d.x;
+        })
+        .attr('height', function (d) {
+          return isHorizontal ? height : d.w;
+        })
+        .attr('width', function (d) {
+          return isHorizontal ? d.w : width;
+        });
+
+      function callPlay(event) {
+        const boundData = event.target.__data__;
+        const mouseChartXCoord = event.clientX - self.chartEl.getBoundingClientRect().left;
+        const mouseChartYCoord = event.clientY - self.chartEl.getBoundingClientRect().top;
+        const wholeBucket = boundData && boundData.x != null;
+
+        // the min and max that the endzones start in
+        const min = isHorizontal ? leftEndzone.w : rightEndzone.w;
+        const max = isHorizontal ? rightEndzone.x : leftEndzone.x;
+
+        // bounds of the cursor to consider
+        let xLeft = isHorizontal ? mouseChartXCoord : mouseChartYCoord;
+        let xRight = isHorizontal ? mouseChartXCoord : mouseChartYCoord;
+        if (wholeBucket) {
+          xLeft = xScale(boundData.x);
+          xRight = xScale(xAxis.addInterval(boundData.x));
+        }
+
+        return {
+          wholeBucket: wholeBucket,
+          touchdown: min > xLeft || max < xRight
+        };
+      }
+
+      function textFormatter() {
+        return touchdownTmpl(callPlay(d3.event));
+      }
+
+      const endzoneTT = new Tooltip('endzones', this.handler.el, textFormatter, null);
+      this.tooltips.push(endzoneTT);
+      endzoneTT.order = 0;
+      endzoneTT.showCondition = function inEndzone() {
+        return callPlay(d3.event).touchdown;
+      };
+      endzoneTT.render()(svg);
+    };
+
+    draw() {
+      // todo: do we need to handle width and height here ?
```

Comment:
Might want to uppercase `TODO` here for consistency with other TODOs in the code.